src/recorder/jamcontroller.cpp

```diff
@@ -169,6 +174,19 @@ void CJamController::SetRecordingDir ( QString newRecordingDir,
     }
 }
 
+void CJamController::OnRecordingFailed ( QString error )
+{
+    if ( !bEnableRecording )
+    {
+        // Recording has already been stopped, possibly
+        // by a previous OnRecordingFailed call from another client/thread.
+        return;
+    }
+    strRecorderErrMsg = error;
+    qWarning() << "Could not start recording:" << error;
+    pServer->SetEnableRecording ( false );
```

Comment:
Might be worth a comment saying what we're doing here.  Does this just stop the _current_ recording or turn off the recorder until it's turned back on?

Reply:
Added a comment.